global_defs.scad

```diff
@@ -42,15 +44,15 @@ pp4_colour      = is_undef($pp4_colour)      ? "darkorange" : $pp4_colour;// pri
 
 
 // Minimum wall is about two filaments wide but we extrude it closer to get better bonding
-squeezed_wall = $preview ? 2 * extrusion_width - layer_height * (1 - PI / 4)
-                         : extrusion_width - layer_height / 2 + nozzle / 2 + extrusion_width / 2;
+squeezed_wall = $preview ? 2 * extrusion_width() - layer_height() * (1 - PI / 4)
+                         : extrusion_width() - layer_height() / 2 + nozzle / 2 + extrusion_width() / 2;
 
 inf = 1e10;      // very big
 eps = 1/128;     // small fudge factor to stop CSG barfing on coincident faces.
 $fa = 6;
-$fs = extrusion_width / 2;
+$fs = extrusion_width() / 2;
```

Comment:
How can I make this dynamic? I can't turn $fs into a function.

Reply:
If you want to override $fa and $fs from the callers, then set these at the top of externally used modules with something like $fs = get_fs(); and put whatever logic you want in that function.  I specifically didn't edit these because you mentioned that your desired behavior for $fa and $fs was disrupted in some parts of the code for some usages when you switched from use to include to try to address the issue without using functions.  But this is not necessary, and you could go back to use if you like.  Setting them via a function call allows full control and makes very clear exactly where and when the caller's $fa and $fs values are being overridden, with what behavior, and where they are not overridden.  Implementing this part I will have to leave to you because the full desired behavior for this was not clarified, but that is how to accomplish it.

Reply:
I have no idea what you mean. I want `$fs` to equal `extrusion_width() / 2` dynamically as it did before. I.e. it responds to $extrusion_width set at file scope or on the stack.